File: js/src/maplestory/api/dto/characterBeautyEquipmentDto.ts
import {CharacterBeautyEquipmentHairDto} from "./characterBeautyEquipmentHairDto";
import {CharacterBeautyEquipmentFaceDto} from "./characterBeautyEquipmentFaceDto";
import {CharacterBeautyEquipmentDtoBody} from "../response/characterBeautyEquipmentDtoBody";

/**
 * 캐릭터 헤어, 성형, 피부 정보
 */
class CharacterBeautyEquipmentDto {

    /**
     * 조회 기준일
     */
    date: Date;

    /**
     * 캐릭터 성별
     */
    characterGender: string;

    /**
     * 캐릭터 직업
     */
    characterClass: string;

    /**
     * 캐릭터 헤어 정보<br>
     * (제로인 경우 알파, 엔젤릭버스터인 경우 일반 모드)
     */
    characterHair: CharacterBeautyEquipmentHairDto;

    /**
     * 캐릭터 성형 정보<br>
     * (제로인 경우 알파, 엔젤릭버스터인 경우 일반 모드)
     */
    characterFace: CharacterBeautyEquipmentFaceDto;

    /**
     * 피부 명<br>
     * (제로인 경우 알파, 엔젤릭버스터인 경우 일반 모드)
     */
    characterSkinName: string;

    /**
     * 제로인 경우 베타, 엔젤릭버스터인 경우 드레스 업 모드에 적용 중인 헤어 정보
     */
    additionalCharacterHair: CharacterBeautyEquipmentHairDto;

    /**
     * 제로인 경우 베타, 엔젤릭버스터인 경우 드레스 업 모드에 적용 중인 성형 정보
     */
    additionalCharacterFace: CharacterBeautyEquipmentFaceDto;

    /**
     * 제로인 경우 베타, 엔젤릭버스터인 경우 드레스 업 모드에 적용 중인 피부 명
     */
    additionalCharacterSkinName: string;

    constructor(obj: CharacterBeautyEquipmentDtoBody) {
        const {
            date,
            character_gender,
            character_class,
            character_hair,
            character_face,
            character_skin_name,
            additional_character_hair,
            additional_character_face,
            additional_character_skin_name,
        } = obj;

        this.date = new Date(date);
        this.characterGender = character_gender;
        this.characterClass = character_class;
        this.characterHair = new CharacterBeautyEquipmentHairDto(character_hair);
        this.characterFace = new CharacterBeautyEquipmentFaceDto(character_face);
        this.characterSkinName = character_skin_name;
        this.additionalCharacterHair = new CharacterBeautyEquipmentHairDto(additional_character_hair);
        this.additionalCharacterFace = new CharacterBeautyEquipmentFaceDto(additional_character_face);
        this.additionalCharacterSkinName = additional_character_skin_name;
    }
}

export {CharacterBeautyEquipmentDto};

File: java/src/main/java/dev/spiralmoon/maplestory/api/dto/CharacterBeautyEquipmentDTO.java
package dev.spiralmoon.maplestory.api.dto;

import com.google.gson.annotations.SerializedName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.ToString;

/**
 * 캐릭터 헤어, 성형, 피부 정보
 */
@AllArgsConstructor
@Data
@ToString
public class CharacterBeautyEquipmentDTO {

    /**
     * 조회 기준일 (KST)
     */
    @SerializedName("date")
    private String date;

    /**
     * 캐릭터 성별
     */
    @SerializedName("character_gender")
    private String characterGender;

    /**
     * 캐릭터 직업
     */
    @SerializedName("character_class")
    private String characterClass;

    /**
     * 캐릭터 헤어 정보
     * (제로인 경우 알파, 엔젤릭버스터인 경우 일반 모드)
     */
    @SerializedName("character_hair")
    private CharacterBeautyEquipmentHairDTO characterHair;

    /**
     * 캐릭터 성형 정보
     * (제로인 경우 알파, 엔젤릭버스터인 경우 일반 모드)
     */
    @SerializedName("character_face")
    private CharacterBeautyEquipmentFaceDTO characterFace;

    /**
     * 피부 명
     * (제로인 경우 알파, 엔젤릭버스터인 경우 일반 모드)
     */
    @SerializedName("character_skin_name")
    private String characterSkinName;

    /**
     * 드레스 업 모드에 적용 중인 헤어 정보
     * (제로인 경우 베타, 엔젤릭버스터인 경우 드레스 업 모드)
     */
    @SerializedName("additional_character_hair")
    private CharacterBeautyEquipmentHairDTO additionalCharacterHair;

    /**
     * 드레스 업 모드에 적용 중인 성형 정보
     * (제로인 경우 베타, 엔젤릭버스터인 경우 드레스 업 모드)
     */
    @SerializedName("additional_character_face")
    private CharacterBeautyEquipmentFaceDTO additionalCharacterFace;

    /**
     * 드레스 업 모드에 적용 중인 피부 명
     * (제로인 경우 베타, 엔젤릭버스터인 경우 드레스 업 모드)
     */
    @SerializedName("additional_character_skin_name")
    private String additionalCharacterSkinName;
}
